### DISCUSSION TITLE
Adapt to different resolutions of input datasets.

### DISCUSSION THREAD

**Discussion by longbiao1993** [WAM2layers/WAM2layers#277] [Q&A] [answered]
Hi, I want to apply this code to my simulated data (1°). And I find the default resolutions is 0.25°(ERA5), but ERA5 can be downloaded in other resolutions.  If the code can be used for other resolutions, I can adjust my data to ERA5 ECMWF table by python or NCL. I believe if this can be done, the code will be used by many others including cmip6 and other model outputing. Please give me some suggestion  to adjust the code in 1° resolution.

**Comment by Peter9192** [comment, marked as answer]
Hello @longbiao1993, and welcome to the WAM2layers repo! I've moved your issue to the discussion forum, which is the best place for Q&A like this one. 

Your use case is exactly why we made a seperation between pre-processing and tracking. The preprocessing step is meant to make any dataset ready for use with the WAM2layers core tracking code. 

Currently, we have only one pre-processing script, for era5 at 0.25 degrees. Adding more pre-processing scripts, for CMIP6 among others, is on our radar, though it might take a while before we get to it. 

If you want to have a go at it, please have a look at the documentation [here](https://wam2layers.readthedocs.io/en/latest/userguide.html#pre-processing), where we describe in detail what the pre-processing script needs to do.

We'd appreciate it if you can keep us posted on any developments on your side. Perhaps we can incorporate your changes in the tool, so others can also benefit from them. We'd be happy to help with that.

PS: to make/keep the WAM2layers GitHub a nice and welcoming place, we find it helps to know the people we are engaging with. Would you mind briefly introducing yourself [here](https://github.com/WAM2layers/WAM2layers/discussions/categories/users-of-wam2layers)?
